lib/execution_engine2/db/models/models.py

```diff
@@ -310,8 +311,9 @@ class Job(Document):
     job_input = EmbeddedDocumentField(JobInput, required=True)
     job_output = DynamicField()
     condor_job_ads = DynamicField()
+    child_jobs = ListField()
```

Comment:
It looks like there isn't a precedent for specifying dependencies between attributes, i.e. if `batch_job` is true, `child_jobs` must be populated :\

Reply:
Not sure what you mean by precedent, but the batch job needs to be submitted and saved in order to pass its ID to the child jobs, so they can set that ID as the parent. So we have to save the record twice, or else we have to save the children twice.